src/sidebar/store/modules/selection.js

```diff
@@ -1,42 +1,34 @@
-/**
- * This module handles the state affecting the visibility and presence of
- * annotations and threads in the UI.
- */
+import { createSelector } from 'reselect';
+
+import * as metadata from '../../helpers/annotation-metadata';
+import { countIf, trueKeys, toTrueMap } from '../../util/collections';
+import { createStoreModule, makeAction } from '../create-store';
 
 /**
  * @typedef {import('../../../types/api').Annotation} Annotation
+ * @typedef {import('../../../types/config').SidebarSettings} SidebarSettings
  * @typedef {import("../../../types/sidebar").TabName} TabName
  */
 
-/** @typedef {'Location'|'Newest'|'Oldest'} SortKey */
-
 /**
- * @typedef State
- * @prop {Record<string, boolean>} expanded
- * @prop {Record<string, boolean>} forcedVisible
- * @prop {Record<string, boolean>} selected
- * @prop {SortKey} sortKey
- * @prop {'annotation'|'note'|'orphan'} selectedTab
+ * @typedef {Record<string, boolean>} BooleanMap
```

Comment:
Perhaps we should consider allowing the use of `Map` and `Set` in store modules in future, in an immutable way. That's out of scope for this PR though.